src/go/rpk/pkg/cli/cmd/cluster/config/importconfig/import.go

```diff
@@ -0,0 +1,165 @@
+	out.MaybeDie(err, "Error parsing edited config: %v", err)
+
+	// Calculate deltas
+	upsert := make(map[string]interface{})
+	remove := make([]string, 0)
+	for k, v := range readbackConfig {
+		oldVal, haveOldVal := old_config[k]
+		if meta, ok := schema[k]; ok {
+			// For numeric types need special handling because
+			// yaml encoding will see '1' as an integer, even
+			// if it is given as the value for a floating point
+			// ('number') config property, and vice versa.
+			if meta.Type == "integer" {
+				switch v_float := v.(type) {
+				case float64:
+					v = int(v_float)
+				}
+
+				if oldVal != nil {
+					oldVal = int(oldVal.(float64))
+				}
+			} else if meta.Type == "number" {
+				switch x := v.(type) {
+				case int:
+					v = float64(x)
+				}
+			} else if meta.Type == "array" && meta.Items.Type == "string" {
+				if v != nil {
+					v = loadStringArray(v.([]interface{}))
+				}
+				if oldVal != nil {
+					oldVal = loadStringArray(oldVal.([]interface{}))
+				}
+			}
+
+			// For types that aren't numberic or array, pass them through as-is
+		}
+
+		if haveOldVal {
+			// If value changed, add it to list of updates
+			// DeepEqual because values can be slices
+			if !reflect.DeepEqual(oldVal, v) {
+				log.Infof("Changed property %s: %v -> %v", k, oldVal, v)
+				upsert[k] = v
+			}
+		} else {
+			// Present in input but not original config, insert
+			log.Infof("New property %s", k)
+			upsert[k] = v
+		}
+	}
+
+	for k := range old_config {
+		if _, found := readbackConfig[k]; !found {
+			meta, in_schema := schema[k]
+			if !in_schema {
+				continue
+			}
+
+			if !all && meta.Visibility == "tunable" {
+				continue
+			}
+			log.Infof("Clearing property %s (was %v)", k, old_config[k])
+			remove = append(remove, k)
+		}
+	}
+
+	if len(upsert) == 0 && len(remove) == 0 {
+		log.Infof("No changes were made")
+		return nil
+	}
+
+	// PUT to admin API
+	result, err := client.PatchClusterConfig(upsert, remove)
+	out.MaybeDie(err, "Error setting config: %v", err)
+	fmt.Printf("Updated config, new config version %d\n", result.ConfigVersion)
+
+	return nil
+}
+
+func NewCommand(fs afero.Fs, all *bool) *cobra.Command {
+	var filename string
+	cmd := &cobra.Command{
+		Use:   "import",
+		Short: "Import cluster configuration from a file",
+		Long: `Import cluster-wide configuration properties
+
+Import configuration from a YAML file, usually generated with corresponding 'export' command.
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			client, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			// GET the schema
+			schema, err := client.ClusterConfigSchema()
+			out.MaybeDie(err, "unable to query config schema: %v", err)
+
+			// GET current config
+			var current_config admin.Config
+			current_config, err = client.Config()
```

Comment:
Do we need to be worried about a concurrent update changing the "old" value here before `ImportConfig` finishes? I don't think we need to be, because concurrent updates are basically non-deterministic anyway, but maybe worth mentioning in the long text.

Reply:
It shouldn't break anything: the list of changes generated by the old/new diff gets validated/trimmed again server side (at least, it does in a subsequent PR...) so if someone decides to overlap edits they'll get a sane result.  I've fleshed out the long text to say that this command is reading and calculating a delta, which should be enough information for anyone wondering "what if I..." about overlapping changes.

We could make this stuff transactional by passing an old version number around and asserting it on the server side when handling an update, but I think it would be overkill.